File: emby-lib/src/main/java/us/nineworlds/serenity/emby/model/Part.java
package us.nineworlds.serenity.emby.model;

import java.util.List;
import us.nineworlds.serenity.common.media.model.IPart;
import us.nineworlds.serenity.common.media.model.IStream;

public class Part implements IPart {

  private String key;
  private String filename;
  private String container;
  private List<IStream> streams;

  @Override public List<IStream> getStreams() {
    return streams;
  }

  @Override public void setStreams(List<IStream> streams) {
    this.streams = streams;
  }

  @Override public String getContainer() {
    return container;
  }

  @Override public void setContainer(String container) {
    this.container = container;
  }

  @Override public String getKey() {
    return key;
  }

  @Override public void setKey(String key) {
    this.key = key;
  }

  @Override public String getFilename() {
    return filename;
  }

  @Override public void setFilename(String filename) {
    this.filename = filename;
  }
}

File: jellyfin-lib/src/main/kotlin/us/nineworlds/serenity/jellyfin/model/Part.kt
package us.nineworlds.serenity.jellyfin.model

import us.nineworlds.serenity.common.media.model.IPart
import us.nineworlds.serenity.common.media.model.IStream

class Part : IPart {
    private var _key: String? = null
    private var _filename: String? = null
    private var _container: String? = null
    private var _streams: List<IStream>? = null

    override fun getStreams(): List<IStream>? = _streams

    override fun setStreams(streams: List<IStream>?) {
        _streams = streams
    }

    override fun getContainer(): String? = _container

    override fun setContainer(container: String?) {
        _container = container
    }

    override fun getKey(): String? = _key

    override fun setKey(key: String?) {
        _key = key
    }

    override fun getFilename(): String? = _filename

    override fun setFilename(filename: String?) {
        _filename = filename
    }
}
